src/mongo/bson/bsonobjbuilder.h

```diff
@@ -752,6 +785,7 @@ namespace mongo {
         BufBuilder &subarrayStart() { return _b.subarrayStart( num() ); }
 
         BSONArrayBuilder& appendRegex(const StringData& regex, const StringData& options = "") {
+            uassert(0, "regex cannot contain null bytes", regex.find('\0') == std::string::npos);
```

Comment:
I wanted the error message to be clear that it's a regex as opposed to a fieldName.